### PR TITLE
Support Guitar Pro 7 track volume

### PR DESCRIPTION
**Pull request by jordanske**
Added support for importing the track volumes from Guitar Pro 7 file format.

### CODE REVIEW

**Comment by Danielku15** [comment]
Can you please first open an issue describing in detail what is the missing feature. It is not really comfortable to just get some code provided without deeper insight what you're trying to solve.

**Comment by jordanske** [comment]
Sorry, this is the first time I contribute something on Github. 
I have created a new issue with an explanation. #446 

Also, am I correct to assume that the web check error is due to the test expecting a different volume than set in Guitar Pro? Which gets imported correctly now from the grace-beats.gp.

**Comment by Danielku15** [comment]
@jordanske That's fine, thanks for creating the issue describing the bug you're solving. 😉  The failing tests rather indicate that some test expectations need also to be adapted. The old code will "wrongly" keep the default volume which is checked in `MidiFileGenerator.test.ts` Line 201 and 210. With your change we now read different values from the file and the expected events need update.

File: src/importer/GpifParser.ts [hunk 1]
```diff
@@ -977,6 +980,44 @@ export class GpifParser {
...
+        }
+    }
+
+    private parseChannelStripParameters(track: Track, node: XmlNode): void {
+        if (node.firstChild && node.firstChild.value) {
+            let parameters = node.firstChild.value.split(' ');
+            for (let i = 0; i < parameters.length; i++) {
+                let value: number = parseInt(parameters[i]);
+                switch (i) {
+                    case 12:
```
Comment (Danielku15) [inline src/importer/GpifParser.ts:1013]: You can simplify this to something similar to: 
```ts
const parameters = node.innerText.split(' ');
if(parameters.length >= 13) {
    track.playbackInfo.volume = Math.floor(parameters[12] * 16) 
}
```

**Review by Danielku15 — CHANGES_REQUESTED** [review]
The change looks good in general. I added one small remark on simplifying the parsing of the channel strip parameters. Could you ensure (like described above) that the tests are updated with the right new values? Also it would be great if you can add some dedicated test to this new feature. e.g. One new testfile with multiple tracks having different volumes, and then a new test checking the expected volumes.

**Comment by jordanske** [comment]
I simplified the parser code, and fixed the Grace Beats test.
I also made a new .gp file with multiple track and a new test that checks if the values are correctly imported.

Should I make a new pull request (and a new issue) for importing balance values, or can I implement it in this one?

**Comment by Danielku15** [comment]
I would recommend adding the balance information directly as part of this PR. I guess it is just one of these parameters in the list so it should be fairly straight forward without big impact to add it here.

**Comment by jordanske** [comment]
I added the balance and also made a new test for it.

**Comment by Danielku15** [comment]
Thanks a lot for this contribution 😃

**Comment by jordanske** [comment]
Awesome! 
You're welcome and thanks!

**Pull request merged**